backend/src/routes/getHistoricalBalance.ts

```diff
@@ -0,0 +1,47 @@
+import { eq, sql } from 'drizzle-orm';
+import type { FastifyInstance } from 'fastify';
+import { ADDRESS_REGEX } from '.';
+import { usdcBalance } from '../db/schema';
+
+export function getHistoricalBalanceRoute(fastify: FastifyInstance) {
+  fastify.get('/get_historical_balance', async (request, reply) => {
+    const { address } = request.query as { address?: string };
+
+    if (!address) {
+      return reply.status(400).send({ error: 'Address is required' });
+    }
+
+    // Validate address format
+    if (!ADDRESS_REGEX.test(address)) {
+      return reply.status(400).send({ error: 'Invalid address format' });
+    }
+
+    try {
+      const subquery = sql`
+        SELECT address, MAX(block_timestamp) AS max_timestamp
+        FROM ${usdcBalance}
+        WHERE address = ${address} AND block_timestamp >= NOW() - INTERVAL '30 days'
+        GROUP BY address, DATE(block_timestamp)
+      `;
```

Comment:
```suggestion
      const subquery = fastify.db
        .select({
          address: usdcBalance.address,
          maxTimestamp: sql`MAX(${usdcBalance.blockTimestamp})`,
        })
        .from(usdcBalance)
        .where(
          and(
            eq(usdcBalance.address, address),
            gte(usdcBalance.blockTimestamp, sql`NOW() - INTERVAL '30 days'`),
          ),
        )
        .groupBy(usdcBalance.address, sql`DATE(${usdcBalance.blockTimestamp})`);
```

Can be rewritten in pure drizzle for type safety. Nothing else needs changing from my test.

Reply:
Yea, found a way to get it to work with pure drizzle. Thank you!